docs/tutorial/native-code-and-electron-cpp-linux.md

```diff
@@ -0,0 +1,1595 @@
+# Native Code and Electron: C++ (Linux)
+
+This tutorial builds on the [general introduction to Native Code and Electron](./native-code-and-electron.md) and focuses on creating a native addon for Linux using C++ and GTK3. To illustrate how you can embed native Linux code in your Electron app, we'll be building a basic native GTK3 GUI that communicates with Electron's JavaScript.
+
+Specifically, we'll be using GTK3 for our GUI interface, which provides:
+
+* A comprehensive set of UI widgets like buttons, entry fields, and lists
+* Cross-desktop compatibility across various Linux distributions
+* Integration with the native theming and accessibility features of Linux desktops
+
+> [!NOTE]
+> We specifically use GTK3 because that's what Chromium (and by extension, Electron) uses internally. Using GTK4 would cause runtime conflicts since both GTK3 and GTK4 would be loaded in the same process. If and when Chromium upgrades to GTK3, you will likely be able to easily upgrade your native code to GTK4, too.
```

Comment:
```suggestion
> We specifically use GTK3 because that's what Chromium (and by extension, Electron) uses internally. Using GTK4 would cause runtime conflicts since both GTK3 and GTK4 would be loaded in the same process. If and when Chromium upgrades to GTK4, you will likely be able to easily upgrade your native code to GTK4, too.
```